recipes/fast-dds/all/conanfile.py

```diff
@@ -197,13 +198,15 @@ def package_info(self):
         self.cpp_info.components["fastrtps"].build_modules["cmake_find_package"] = [self._module_file_rel_path]
         self.cpp_info.components["fastrtps"].build_modules["cmake_find_package_multi"] = [self._module_file_rel_path]
         # component fast-discovery
-        self.cpp_info.components["fast-discovery"].name = "fast-discovery"
-        self.cpp_info.components["fast-discovery"].bindirs = ["bin"]
+        self.cpp_info.components["fast-discovery-server"].names["cmake_find_package"] = "fast-discovery-server"
+        self.cpp_info.components["fast-discovery-server"].names["cmake_find_multi_package"] = "fast-discovery-server"
+        self.cpp_info.components["fast-discovery-server"].bindirs = ["bin"]
         bin_path = os.path.join(self.package_folder, "bin")
-        self.output.info("Appending PATH env var for fast-dds::fast-discovery with : {}".format(bin_path)),
+        self.output.info("Appending PATH env var for fast-dds::fast-discovery-server with : {}".format(bin_path)),
         self.env_info.PATH.append(bin_path)
         # component tools
-        self.cpp_info.components["tools"].name = "tools"
+        self.cpp_info.components["tools"].names["cmake_find_package"] = "tools"
```

Comment:
The name here is a little generic... maybe?

```suggestion
        self.cpp_info.components["fast-dds-tools"].names["cmake_find_package"] = "tools"
```